src/main/java/com/google/sps/servlets/DataFromDatabaseServlet.java

```diff
@@ -33,59 +35,67 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType(constants.TEXT_TYPE);
     List<Table> tables = new ArrayList<>();
 
-     for (String table : selectedTables) {
+    for (String table : selectedTables) {
       String columnQuery = constants.SCHEMA_INFO_SQL + table + constants.ENDING_APOSTROPHE;
  
       try (ResultSet resultSet =
         dbClient.singleUse().executeQuery(Statement.of(columnQuery))) {
         List<Schema> schemas = new ArrayList<>();
- 
         while (resultSet.next()) {
           schemas.add(createSchema(resultSet));
         }
  
         // No columns -> throw error
         if (schemas.size() == 0) {
-          throw new RuntimeException("Table is empty.");
+          throw new RuntimeException(constants.EMPTY_TABLE_ERROR);
         }
  
-        Table tableObject = new Table(table);
-        String query = constructQueryString(schemas, tableObject, table);
-        executeTableQuery(tableObject, query, schemas);
+        Table.Builder tableBuilder = Table.builder().setName(table);
+
+        ImmutableList.Builder<String> columnNamesBuilder = new ImmutableList.Builder<String>();
+        Statement queryStatement = constructQueryStatement(schemas, columnNamesBuilder, table);
+        ImmutableList<String> columnNames = columnNamesBuilder.build();
+        tableBuilder.setColumns(columnNames);
+
+        executeTableQuery(tableBuilder, queryStatement, schemas);
+        
+        Table tableObject = tableBuilder.build();
         tables.add(tableObject);
       }
     }
-
     String json = new Gson().toJson(tables);
     response.getWriter().println(json);
   }
 
   private Schema createSchema(ResultSet resultSet) {
     String columnName = resultSet.getString(0);
     String schemaType = resultSet.getString(1);
+
+    // Remove length in parens (i.e. STRING(MAX) --> STRING)
     int indexOfOpeningParen = schemaType.indexOf("(");
     if (indexOfOpeningParen >= 0) {
       schemaType = schemaType.substring(0, indexOfOpeningParen);
     }
     String nullableString = resultSet.getString(2);
     boolean nullable = false;
-    if (nullableString.toLowerCase().equals("true")) {
+    if (nullableString.toLowerCase().equals(constants.TRUE)) {
       nullable = true;
     }
     return Schema.create(columnName, schemaType, nullable);
   }
 
-  //TODO (issue 15): use StringBuilder instead
-  private String constructQueryString(List<Schema> schemas, Table tableObject, String table) {
-    String query = constants.SELECT;
+  private Statement constructQueryStatement(List<Schema> schemas, ImmutableList.Builder<String> columnNamesBuilder, String table) {
```

Comment:
Thank you for this comment - that's really helpful to learn. I think it would make sense to get rid of the columnNames list altogether, since the names are already in ColumnSchema.

Reply:
For now I'm going to at least separate out the different methods. In a separate branch, I've already put the list of ColumnSchemas into the Table. So I'll get rid of the columnNames list in that branch for a new PR.